code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ManualCarouselNotificationBuilderTest.kt

```diff
@@ -185,6 +223,74 @@ class ManualCarouselNotificationBuilderTest {
         )
     }
 
+    @Test
+    fun `test to verify notification with action name MANUAL_CAROUSEL_LEFT_CLICKED is having correct center image index`() {
+        every { cacheImages(any()) } answers { 5 }
+        every { getCachedImage(any()) } answers { mockkClass(Bitmap::class) }
+        val mockBundle = MockCarousalTemplateDataProvider.getMockedBundleWithManualCarouselData()
+        val data = IntentData(mockBundle, PushTemplateConstants.IntentActions.MANUAL_CAROUSEL_LEFT_CLICKED)
+        val mcPushTemplate = CarouselPushTemplate(data) as ManualCarouselPushTemplate
+        val centerIndex = mcPushTemplate.centerImageIndex
+        ManualCarouselNotificationBuilder.construct(
+            context,
+            mcPushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        assertTrue(mcPushTemplate.centerImageIndex == (centerIndex - 1 + 5) % 5)
+    }
+
+    @Test
+    fun `test to verify notification with action name MANUAL_CAROUSEL_RIGHT_CLICKED is having correct center image index`() {
+        every { cacheImages(any()) } answers { 5 }
+        every { getCachedImage(any()) } answers { mockkClass(Bitmap::class) }
+        val mockBundle = MockCarousalTemplateDataProvider.getMockedBundleWithManualCarouselData()
+        val data = IntentData(mockBundle, PushTemplateConstants.IntentActions.MANUAL_CAROUSEL_RIGHT_CLICKED)
+        val mcPushTemplate = CarouselPushTemplate(data) as ManualCarouselPushTemplate
+        val centerIndex = mcPushTemplate.centerImageIndex
+        ManualCarouselNotificationBuilder.construct(
+            context,
+            mcPushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        assertTrue(mcPushTemplate.centerImageIndex == (centerIndex + 1) % 5)
+    }
+
+    @Test
+    fun `test to verify notification with action name FILMSTRIP_LEFT_CLICKED is having correct center image index`() {
+        every { cacheImages(any()) } answers { 5 }
+        every { getCachedImage(any()) } answers { mockkClass(Bitmap::class) }
+        val mockBundle = MockCarousalTemplateDataProvider.getMockedBundleWithManualCarouselData()
+        val data = IntentData(mockBundle, PushTemplateConstants.IntentActions.FILMSTRIP_LEFT_CLICKED)
+        val mcPushTemplate = CarouselPushTemplate(data) as ManualCarouselPushTemplate
+        val centerIndex = mcPushTemplate.centerImageIndex
+        ManualCarouselNotificationBuilder.construct(
+            context,
+            mcPushTemplate,
+            trackerActivityClass,
+            broadcastReceiverClass
+        )
+        assertTrue(mcPushTemplate.centerImageIndex == (centerIndex - 1 + 5) % 5)
```

Comment:
we can update this to `assertEquals(0, centerIndex)`. filmstrip carousels have a default center index of 1 and a left click would make it 0.

Reply:
done @rymorale